ClientApp/src/components/pages/EmployeeDirectoryPage/EmployeeDirectoryPage.tsx

```diff
@@ -1,43 +1,77 @@
 import {
   AsyncComponent,
+  EditTeamModal,
   ErrorModal,
   Header,
   PermissionsWrapper,
-  useErrorHandling,
 } from "components";
 import { chain, Dictionary } from "lodash";
-import React, { useEffect, useState } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 import Button from "react-bootstrap/Button";
 import Col from "react-bootstrap/Col";
 import Row from "react-bootstrap/Row";
-import { getAllEmployees, useRequireLogin } from "services";
+import { getAllEmployees, useErrorHandling, useRequireLogin } from "services";
+import { useModal } from "utilities";
 import { EmployeeOverview, PermissionType } from "_generated";
 import "./EmployeeDirectoryPage.scss";
 import { EmployeeDirectoryTable } from "./EmployeeDirectoryTable";
 
 export const EmployeeDirectoryPage: React.FC = (): JSX.Element => {
-  const [isLoading, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
   const [employees, setEmployees] = useState<EmployeeOverview[]>([]);
   const { error, handleError } = useErrorHandling();
+  const {
+    isModalShowing,
+    setIsModalShowing,
+    setModalData,
+    modalData,
+  } = useModal<EmployeeOverview[]>();
   useRequireLogin();
 
+  /**
+   * Used when sorting and grouping employees.
+   */
+  const getManagerName = (e: EmployeeOverview): string => {
+    if (e.manager?.employeeId === e.employeeId) return "Unassigned";
+    return e.manager?.fullName || "";
+  };
```

Comment:
Semantically I think this would make more sense:
```suggestion
  const getManagerName = (e: EmployeeOverview): string | undefined => {
    if (e.manager?.employeeId === e.employeeId) return undefined;
    return e.manager?.fullName || undefined;
  };
```
Not that it would ever happen, but this would fix an edge case where someone's actual name is "Unassigned". More importantly, you wouldn't have to hardcode the string "Unassigned" in other places where you're checking for an employee's manager.